src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -515,154 +266,119 @@ public static void writeRandomSymbolValues(int size, String format, String path)
-            case INT:
-                long intValue = WriteRandomIonValues.constructInt();
-                writer.writeInt(intValue);
+                writer.writeDecimal(WriteRandomIonValues.constructDecimal(constraintStruct));
                 break;
             case TIMESTAMP:
-                if (timestampTemplate != null) {
-                    try (IonReader templateReader = IonReaderBuilder.standard().build(timestampTemplate)) {
-                        if (templateReader.next() != IonType.LIST) throw new IllegalStateException("Please provide a list type");
-                        templateReader.stepIn();
-                        while (templateReader.next() != null) {
-                            Timestamp value;
-                            if (templateReader.getType() == IonType.TIMESTAMP) {
-                                value = templateReader.timestampValue();
-                            } else {
-                                throw new IllegalStateException("Please provide timestamp in the template list");
-                            }
-                            Timestamp.Precision precision = value.getPrecision();
-                            Timestamp timestamp = WriteRandomIonValues.writeTimestamp(precision, value);
-                            writer.writeTimestamp(timestamp);
-                        }
-                        if (templateReader.next() != null) throw new IllegalStateException("Only one template list is needed");
-                    }
-                } else {
-                    int randomIndex = IonSchemaUtilities.parseConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_TIMESTAMP_PRECISION);
-                    Timestamp.Precision precision = PRECISIONS[randomIndex];
-                    Timestamp timestamp = WriteRandomIonValues.constructTimestamp(precision);
-                    writer.writeTimestamp(timestamp);
-                }
+                writer.writeTimestamp(WriteRandomIonValues.constructTimestamp(constraintStruct));
                 break;
             case STRUCT:
-                annotationList = IonSchemaUtilities.getAnnotation(constraintStruct);
-                WriteRandomIonValues.constructIonStruct(constraintStruct, annotationList, writer);
+                WriteRandomIonValues.constructAndWriteIonStruct(constraintStruct, writer);
                 break;
             case LIST:
-                annotationList = IonSchemaUtilities.getAnnotation(constraintStruct);
-                WriteRandomIonValues.constructIonList(writer, constraintStruct, annotationList);
+                WriteRandomIonValues.constructAndWriteIonList(writer, constraintStruct);
+                break;
+            case BLOB:
+                writer.writeBlob(WriteRandomIonValues.constructLobs(constraintStruct));
+                break;
+            case CLOB:
+                writer.writeClob(WriteRandomIonValues.constructLobs(constraintStruct));
                 break;
         }
```

Comment:
We should probably add a `default` that throws an exception when a type that isn't yet supported is encountered.

Reply:
> We should probably add a `default` that throws an exception when a type that isn't yet supported is encountered.

Will be added in the following commit.